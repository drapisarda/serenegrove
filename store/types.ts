export interface Step {
  id: number
  name: string
  duration: number
  file: string
  description: string
  icon: string
  pauseAfter: number
}

// TODO use standard
export interface TimeFormat {
  hours: string
  minutes: string
  seconds: string
}

export interface ToastMessage {
  message: string
  id: number
  style: string
}

export enum ToastStyles {
  Success = 'success',
  Warning = 'warning',
}

export interface RoutineTimeVariationType {
  id: number
  label: string
  modifier: number
}
