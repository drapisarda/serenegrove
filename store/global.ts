import { defineStore } from "pinia";

export interface ToastMessage {
  message: string,
  id: number,
  style: string
}

export enum ToastStyles {
  Success = 'success',
  Warning = 'warning',
}

export const useGlobalStore = defineStore("global", {
  state: () => ({
    version: '0.1' as string,
    toastMessage: {} as ToastMessage,
  }),
  actions: {
    getVersion(): string {
      return this.$state.version;
    },
    getToastMessage(): ToastMessage {
      return this.$state.toastMessage;
    },
    setToastMessage(message: string, style = 'success'): void {
      this.$state.toastMessage.message = message;
      this.$state.toastMessage.id = new Date().getTime();
      this.$state.toastMessage.style = style;
    }
  }
});