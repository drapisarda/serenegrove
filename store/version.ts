import { defineStore } from "pinia";

export const useVersionStore = defineStore("version", {
  state: () => ({
    version: '0.1' as string,
  }),
  actions: {
    getVersion(): string {
      return this.$state.version;
    }
  }
});