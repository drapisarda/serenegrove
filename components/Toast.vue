<template>
  <div class="toast" :class="{'toast--visible': isVisible}">
    <slot>
      <div class="notification" :class="messageClass">
        <p>
          {{ message }}
        </p>
      </div>
    </slot>
  </div>
</template>

<script lang="ts" setup>
import { useGlobalStore, ToastMessage } from "@/store/global";
import { ref, watch } from "vue";

const message = ref('');
const messageClass = ref('');
const isVisible = ref(false);
const { toastMessage } = useGlobalStore();
watch(toastMessage, (newMessage: ToastMessage) => displayMessage(newMessage));

const displayMessage = (newMessage: ToastMessage, duration = 3000) => {
  isVisible.value = true;
  message.value = newMessage.message;
  messageClass.value = `is-${newMessage.style ? newMessage.style : 'success'}`;
  setTimeout(function () { isVisible.value = false }, duration);
}
</script>

<style scoped lang="scss">
@import "@/style/vars.scss";
$spacing: 3rem;

.toast {
  position: fixed;
  width: 100%;
  bottom: $spacing;
  visibility: hidden;

  &.toast--visible {
    z-index: $toastZIndex;
    visibility: visible;
    animation: fadein 0.5s, fadeout 0.5s 2.5s;
  }

  @keyframes fadein {
    from {
      bottom: 0;
      opacity: 0;
    }

    to {
      bottom: $spacing;
      opacity: 1;
    }
  }

  @keyframes fadeout {
    from {
      bottom: $spacing;
      opacity: 1;
    }

    to {
      bottom: 0;
      opacity: 0;
    }
  }
}
</style>