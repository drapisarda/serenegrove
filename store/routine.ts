import { defineStore } from "pinia";

import { useGlobalStore } from "./global";

export interface Step {
  name: string,
  duration: number,
  file: string,
  description: string,
  icon: string,
  pauseAfter: number,
  iconData: StepIconData,
}

export interface StepIconData {
  url: string,
  altDescription: string
}


const debugAudio = '/assets/audio/1.mp3';
const debug = false;

export const useRoutineStore = defineStore("mainRoutine", {
  state: () => ({
    steps: [] as Step[],
    lastEdit: 0 as number,
    version: useGlobalStore().getVersion() as string,
    intro:  { name: 'Intro', duration: 3000, file: '/assets/audio/intro.mp3', description: 'Welcome to your meditation routine', icon: 'logo_only.svg', pauseAfter: 0 } as Step,
    outro:  { name: 'Outro', duration: 3000, file: '/assets/audio/outro.mp3', description: 'Thank you for meditate with SereneGrove', icon: 'logo_only.svg', pauseAfter: 0 } as Step,
    stepsOptions: [
      { name: 'Breath', duration: 3000, file: '/assets/audio/breath.mp3', description: 'Focus your attention on the sensation of the breath, observing its natural flow without trying to control it.', icon: '005-breath.svg' },
      { name: 'Body scan', duration: 5000, file: '/assets/audio/body_scan.mp3', description: 'Systematically bring awareness to different parts of the body, observing physical sensations without judgment.', icon: '003-meditation.svg' },
      { name: 'Sounds around you', duration: 7000, file: '/assets/audio/sounds.mp3', description: 'Direct your attention to the sounds in your environment, observing them as they arise and fade away without analysis.', icon: '004-landscape.svg' },
      { name: 'Mantra', duration: 5000, file: '/assets/audio/mantra.mp3', description: 'Repeat a chosen word or phrase silently in your mind to cultivate inner calm and focused awareness.', icon: '002-shree.svg' },
      { name: 'Bell sound', duration: 3000, file: '/assets/audio/bell.mp3', description: 'Focus your attention on the sound of a bell, observing its duration, resonance, and fading away to develop concentration.', icon: '001-standing-bell.svg' },
    ] as Step[],
  }),
  actions: {
    addStep(step: Step, insertIndex: number) {
      this.$state.steps.splice(insertIndex, 0, step);
      this.lastEdit = Date.now();
    },
    addStepAtTheBottom(step: Step) {
      this.$state.steps.push(step);
      this.lastEdit = Date.now();
    },
    removeStep(removeIndex: number) {
      if (removeIndex < 0) return;
      this.$state.steps.splice(removeIndex, 1);
      this.lastEdit = Date.now();
    },
    moveStep(stepIndex: number, amount: number) {
      const newIndex = (stepIndex + amount);
      if (newIndex < 0 || newIndex >= this.$state.steps.length) return;
      const element = this.$state.steps[stepIndex];
      this.$state.steps.splice(stepIndex, 1);
      this.$state.steps.splice(newIndex, 0, element);
    },
  },
  getters: {
    stepIconsDataMap(state): Map<string, StepIconData> {
      const baseURL = import.meta.env.BASE_URL;
      const stepIconData = new Map<string, StepIconData>();
      this.playerSteps.forEach((step, index) => stepIconData.set(step.name, {
        url: `${baseURL}assets/img/icons/${step.icon}`,
        altDescription: `${step.name} - ${step.description}`.substring(0, 20),
      }));
  
      return stepIconData;
    },
    playerSteps(state): Step[] {
      const fullRoutire  = [state.intro].concat(state.steps).concat([state.outro]);

      if (debug) return fullRoutire.map(step => {
        step.file = debugAudio;
        return step
      })

      return fullRoutire;

    }
  },
  persist: {
    storage: persistedState.localStorage,
  },
});
