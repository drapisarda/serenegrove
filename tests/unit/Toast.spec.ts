import Toast from '@/components/Toast.vue'
import { mount, type VueWrapper } from '@vue/test-utils'
import { expect } from 'vitest'
import { setActivePinia, createPinia } from 'pinia'
import { useGlobalStore } from '@/store/global'
import { ToastStyles } from '@/store/types'

describe('RoutineCarousel component test', async () => {
  let wrapper: VueWrapper

  beforeAll(() => {
    setActivePinia(createPinia())
    vi.useFakeTimers({ shouldAdvanceTime: true })

    wrapper = mount(Toast, {
      props: {},
    })
  })

  afterAll(() => {
    vi.useRealTimers()
  })

  test('RoutineCarousel mounts', () => {
    expect(wrapper.exists()).toBe(true)
    expect(wrapper).toMatchSnapshot()
  })

  test('Toast a message, then hide it', async () => {
    const { setToastMessage } = useGlobalStore()
    setToastMessage(`Lorem Ipsum dolor sit amet`, ToastStyles.Success)
    await wrapper.vm.$nextTick()
    expect(wrapper.find('.toast').classes().includes('toast--visible')).toBe(
      true,
    )

    await vi.advanceTimersByTimeAsync(5000)
    expect(wrapper.find('.toast').classes().includes('toast--visible')).toBe(
      false,
    )
  })

  test('Toast a message, then hide it', async () => {
    const secondMessage = `Lorem Ipsum dolor sit amet Bis`
    const { setToastMessage } = useGlobalStore()
    setToastMessage(`Lorem Ipsum dolor sit amet`, ToastStyles.Success)
    await wrapper.vm.$nextTick()
    setToastMessage(secondMessage, ToastStyles.Success)
    await wrapper.vm.$nextTick()

    expect(wrapper.find('.toast').text()).toBe(secondMessage)
  })
})
