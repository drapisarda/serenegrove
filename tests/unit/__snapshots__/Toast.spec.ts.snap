// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RoutineCarousel component test > RoutineCarousel mounts 1`] = `
VueWrapper {
  "__app": {
    "_component": {
      "name": "VTU_ROOT",
      "render": [Function],
      "setup": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="toast"
      >
        
        <div
          class="toast__content"
        />
        
      </div>
    </div>,
    "_context": {
      "app": [Circular],
      "components": {
        "transition": {
          "name": "transition",
        },
        "transition-group": {
          "name": "transition-group",
        },
      },
      "config": {
        "compilerOptions": {},
        "errorHandler": undefined,
        "globalProperties": {},
        "isNativeTag": [Function],
        "optionMergeStrategies": {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": {},
      "emitsCache": WeakMap {},
      "mixins": [
        {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": {},
      "reload": [Function],
    },
    "_instance": {
      "a": null,
      "accessCache": {},
      "appContext": {
        "app": [Circular],
        "components": {
          "transition": {
            "name": "transition",
          },
          "transition-group": {
            "name": "transition-group",
          },
        },
        "config": {
          "compilerOptions": {},
          "errorHandler": undefined,
          "globalProperties": {},
          "isNativeTag": [Function],
          "optionMergeStrategies": {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": {},
        "emitsCache": WeakMap {},
        "mixins": [
          {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": {
        "VTU_COMPONENT": {},
      },
      "da": null,
      "data": {},
      "devtoolsRawSetupState": {
        "VTU_COMPONENT": RefImpl {
          "__v_isRef": true,
          "__v_isShallow": false,
          "_rawValue": {},
          "_value": {},
          "dep": undefined,
        },
      },
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "_depsLength": 3,
        "_dirtyLevel": 0,
        "_runnings": 0,
        "_shouldSchedule": false,
        "_trackId": 1,
        "active": true,
        "allowRecurse": true,
        "deps": [
          Map {
            [Circular] => 1,
          },
          Map {
            [Circular] => 1,
          },
          Map {
            [Circular] => 1,
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "scheduler": [Function],
        "trigger": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": {},
      "propsDefaults": {},
      "propsOptions": [
        {},
        [],
      ],
      "provides": {},
      "proxy": {
        "VTU_COMPONENT": {},
      },
      "refs": {
        "VTU_COMPONENT": {},
      },
      "render": [Function],
      "renderCache": [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "_active": true,
        "cleanups": [],
        "detached": true,
        "effects": [
          ReactiveEffect {
            "_depsLength": 3,
            "_dirtyLevel": 0,
            "_runnings": 0,
            "_shouldSchedule": false,
            "_trackId": 1,
            "active": true,
            "allowRecurse": true,
            "deps": [
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "scheduler": [Function],
            "trigger": [Function],
          },
        ],
        "parent": undefined,
      },
      "setupContext": null,
      "setupState": {
        "VTU_COMPONENT": {},
      },
      "slots": {},
      "sp": null,
      "subTree": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": {
          "a": null,
          "accessCache": {},
          "appContext": {
            "app": [Circular],
            "components": {
              "transition": {
                "name": "transition",
              },
              "transition-group": {
                "name": "transition-group",
              },
            },
            "config": {
              "compilerOptions": {},
              "errorHandler": undefined,
              "globalProperties": {},
              "isNativeTag": [Function],
              "optionMergeStrategies": {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": {},
            "emitsCache": WeakMap {},
            "mixins": [
              {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": {},
          "ctx": {
            "_pStores": {
              "global": {
                "$dispose": [Function],
                "$id": "global",
                "$onAction": [Function],
                "$patch": [Function],
                "$reset": [Function],
                "$subscribe": [Function],
                "_customProperties": Set {},
                "_hmrPayload": {
                  "actions": {
                    "getToastMessage": [Function],
                    "getVersion": [Function],
                    "setToastMessage": [Function],
                  },
                  "getters": {},
                  "hotState": RefImpl {
                    "__v_isRef": true,
                    "__v_isShallow": false,
                    "_rawValue": {},
                    "_value": {},
                    "dep": undefined,
                  },
                  "state": [
                    "version",
                    "toastMessage",
                  ],
                },
                "_hotUpdate": [Function],
                "_p": {
                  "_a": null,
                  "_e": EffectScope {
                    "_active": true,
                    "cleanups": [],
                    "detached": true,
                    "effects": [],
                    "parent": undefined,
                    "scopes": [
                      EffectScope {
                        "_active": true,
                        "cleanups": [],
                        "detached": false,
                        "effects": [],
                        "index": 0,
                        "parent": [Circular],
                      },
                    ],
                  },
                  "_p": [],
                  "_s": Map {
                    "global" => [Circular],
                  },
                  "install": [Function],
                  "state": RefImpl {
                    "__v_isRef": true,
                    "__v_isShallow": false,
                    "_rawValue": {
                      "global": {
                        "toastMessage": {},
                        "version": "0.5",
                      },
                    },
                    "_value": {
                      "global": {
                        "toastMessage": {},
                        "version": "0.5",
                      },
                    },
                    "dep": undefined,
                  },
                  "use": [Function],
                },
                "getToastMessage": [Function],
                "getVersion": [Function],
                "setToastMessage": [Function],
                "toastMessage": {},
                "version": "0.5",
              },
            },
          },
          "da": null,
          "data": {},
          "devtoolsRawSetupState": {
            "displayInterval": undefined,
            "displayMessage": [Function],
            "isVisible": RefImpl {
              "__v_isRef": true,
              "__v_isShallow": false,
              "_rawValue": false,
              "_value": false,
              "dep": Map {
                ReactiveEffect {
                  "_depsLength": 4,
                  "_dirtyLevel": 0,
                  "_runnings": 0,
                  "_shouldSchedule": false,
                  "_trackId": 1,
                  "active": true,
                  "allowRecurse": true,
                  "deps": [
                    [Circular],
                    Map {
                      [Circular] => 1,
                    },
                    Map {
                      [Circular] => 1,
                    },
                    Map {
                      [Circular] => 1,
                    },
                  ],
                  "fn": [Function],
                  "onTrack": undefined,
                  "onTrigger": undefined,
                  "scheduler": [Function],
                  "trigger": [Function],
                } => 1,
              },
            },
            "message": RefImpl {
              "__v_isRef": true,
              "__v_isShallow": false,
              "_rawValue": "",
              "_value": "",
              "dep": Map {
                ReactiveEffect {
                  "_depsLength": 4,
                  "_dirtyLevel": 0,
                  "_runnings": 0,
                  "_shouldSchedule": false,
                  "_trackId": 1,
                  "active": true,
                  "allowRecurse": true,
                  "deps": [
                    Map {
                      [Circular] => 1,
                    },
                    Map {
                      [Circular] => 1,
                    },
                    Map {
                      [Circular] => 1,
                    },
                    [Circular],
                  ],
                  "fn": [Function],
                  "onTrack": undefined,
                  "onTrigger": undefined,
                  "scheduler": [Function],
                  "trigger": [Function],
                } => 1,
              },
            },
            "messageClass": RefImpl {
              "__v_isRef": true,
              "__v_isShallow": false,
              "_rawValue": "",
              "_value": "",
              "dep": Map {
                ReactiveEffect {
                  "_depsLength": 4,
                  "_dirtyLevel": 0,
                  "_runnings": 0,
                  "_shouldSchedule": false,
                  "_trackId": 1,
                  "active": true,
                  "allowRecurse": true,
                  "deps": [
                    Map {
                      [Circular] => 1,
                    },
                    [Circular],
                    Map {
                      [Circular] => 1,
                    },
                    Map {
                      [Circular] => 1,
                    },
                  ],
                  "fn": [Function],
                  "onTrack": undefined,
                  "onTrigger": undefined,
                  "scheduler": [Function],
                  "trigger": [Function],
                } => 1,
              },
            },
            "toastMessage": {},
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "_depsLength": 4,
            "_dirtyLevel": 0,
            "_runnings": 0,
            "_shouldSchedule": false,
            "_trackId": 1,
            "active": true,
            "allowRecurse": true,
            "deps": [
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
              Map {
                [Circular] => 1,
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "scheduler": [Function],
            "trigger": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": {},
          "exposed": {},
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": {},
          "propsDefaults": {},
          "propsOptions": [
            {},
            [],
          ],
          "provides": {},
          "proxy": {
            "_pStores": {
              "global": {
                "$dispose": [Function],
                "$id": "global",
                "$onAction": [Function],
                "$patch": [Function],
                "$reset": [Function],
                "$subscribe": [Function],
                "_customProperties": Set {},
                "_hmrPayload": {
                  "actions": {
                    "getToastMessage": [Function],
                    "getVersion": [Function],
                    "setToastMessage": [Function],
                  },
                  "getters": {},
                  "hotState": RefImpl {
                    "__v_isRef": true,
                    "__v_isShallow": false,
                    "_rawValue": {},
                    "_value": {},
                    "dep": undefined,
                  },
                  "state": [
                    "version",
                    "toastMessage",
                  ],
                },
                "_hotUpdate": [Function],
                "_p": {
                  "_a": null,
                  "_e": EffectScope {
                    "_active": true,
                    "cleanups": [],
                    "detached": true,
                    "effects": [],
                    "parent": undefined,
                    "scopes": [
                      EffectScope {
                        "_active": true,
                        "cleanups": [],
                        "detached": false,
                        "effects": [],
                        "index": 0,
                        "parent": [Circular],
                      },
                    ],
                  },
                  "_p": [],
                  "_s": Map {
                    "global" => [Circular],
                  },
                  "install": [Function],
                  "state": RefImpl {
                    "__v_isRef": true,
                    "__v_isShallow": false,
                    "_rawValue": {
                      "global": {
                        "toastMessage": {},
                        "version": "0.5",
                      },
                    },
                    "_value": {
                      "global": {
                        "toastMessage": {},
                        "version": "0.5",
                      },
                    },
                    "dep": undefined,
                  },
                  "use": [Function],
                },
                "getToastMessage": [Function],
                "getVersion": [Function],
                "setToastMessage": [Function],
                "toastMessage": {},
                "version": "0.5",
              },
            },
          },
          "refs": {},
          "render": [Function],
          "renderCache": [],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "_active": true,
            "cleanups": [],
            "detached": true,
            "effects": [
              ReactiveEffect {
                "_depsLength": 2,
                "_dirtyLevel": 0,
                "_runnings": 0,
                "_shouldSchedule": false,
                "_trackId": 1,
                "active": true,
                "deps": [
                  Map {
                    [Circular] => 1,
                  },
                  Map {
                    [Circular] => 1,
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "scheduler": [Function],
                "trigger": [Function],
              },
              ReactiveEffect {
                "_depsLength": 4,
                "_dirtyLevel": 0,
                "_runnings": 0,
                "_shouldSchedule": false,
                "_trackId": 1,
                "active": true,
                "allowRecurse": true,
                "deps": [
                  Map {
                    [Circular] => 1,
                  },
                  Map {
                    [Circular] => 1,
                  },
                  Map {
                    [Circular] => 1,
                  },
                  Map {
                    [Circular] => 1,
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "scheduler": [Function],
                "trigger": [Function],
              },
            ],
            "parent": undefined,
          },
          "setupContext": {
            "attrs": {},
            "emit": [Function],
            "expose": [Function],
            "slots": {},
          },
          "setupState": {
            "displayInterval": undefined,
            "displayMessage": [Function],
            "isVisible": false,
            "message": "",
            "messageClass": "",
            "toastMessage": {},
          },
          "slots": {},
          "sp": null,
          "subTree": {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": ,
                "appContext": null,
                "children": [
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": "",
                    "component": null,
                    "ctx": [Circular],
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <div
                      class="toast__content"
                    />,
                    "key": null,
                    "patchFlag": 1,
                    "props": {
                      "class": "toast__content",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 1,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "targetStart": null,
                    "transition": null,
                    "type": "div",
                  },
                ],
                "component": null,
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": ,
                "key": "_default_fb",
                "patchFlag": -2,
                "props": {
                  "key": "_default_fb",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 16,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "targetStart": null,
                "transition": null,
                "type": Symbol(v-fgt),
              },
            ],
            "component": null,
            "ctx": [Circular],
            "dirs": null,
            "dynamicChildren": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": ,
                "appContext": null,
                "children": [
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": "",
                    "component": null,
                    "ctx": [Circular],
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <div
                      class="toast__content"
                    />,
                    "key": null,
                    "patchFlag": 1,
                    "props": {
                      "class": "toast__content",
                    },
                    "ref": null,
                    "scopeId": null,
                    "shapeFlag": 1,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "targetStart": null,
                    "transition": null,
                    "type": "div",
                  },
                ],
                "component": null,
                "ctx": [Circular],
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": ,
                "key": "_default_fb",
                "patchFlag": -2,
                "props": {
                  "key": "_default_fb",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 16,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "targetStart": null,
                "transition": null,
                "type": Symbol(v-fgt),
              },
            ],
            "dynamicProps": null,
            "el": <div
              class="toast"
            >
              
              <div
                class="toast__content"
              />
              
            </div>,
            "key": null,
            "patchFlag": 2,
            "props": {
              "class": "toast",
            },
            "ref": null,
            "scopeId": null,
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "targetStart": null,
            "transition": null,
            "type": "div",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": {
            "__file": "/Users/danielerapisarda/projects/serenegrove/components/Toast.vue",
            "__name": "Toast",
            "components": {},
            "render": [Function],
            "setup": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "ctx": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="toast"
        >
          
          <div
            class="toast__content"
          />
          
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": {
          "ref": "VTU_COMPONENT",
        },
        "ref": {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "targetStart": null,
        "transition": null,
        "type": {
          "__file": "/Users/danielerapisarda/projects/serenegrove/components/Toast.vue",
          "__name": "Toast",
          "components": {},
          "render": [Function],
          "setup": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": {
        "name": "VTU_ROOT",
        "render": [Function],
        "setup": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": {
          "app": [Circular],
          "components": {
            "transition": {
              "name": "transition",
            },
            "transition-group": {
              "name": "transition-group",
            },
          },
          "config": {
            "compilerOptions": {},
            "errorHandler": undefined,
            "globalProperties": {},
            "isNativeTag": [Function],
            "optionMergeStrategies": {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": {},
          "emitsCache": WeakMap {},
          "mixins": [
            {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "ctx": null,
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="toast"
        >
          
          <div
            class="toast__content"
          />
          
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "targetStart": null,
        "transition": null,
        "type": {
          "name": "VTU_ROOT",
          "render": [Function],
          "setup": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": {
      "compilerOptions": {},
      "errorHandler": undefined,
      "globalProperties": {},
      "isNativeTag": [Function],
      "optionMergeStrategies": {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "runWithContext": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.4.36",
  },
  "__setProps": [Function],
  "cleanUpCallbacks": [
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
    [Function],
  ],
  "componentVM": {},
  "isDisabled": [Function],
  "rootVM": {
    "VTU_COMPONENT": {},
  },
  "wrapperElement": <div
    class="toast"
  >
    
    <div
      class="toast__content"
    />
    
  </div>,
}
`;
